### PR TITLE
Log response payload for better troubleshooting

### PR DESCRIPTION
**Pull request by sungkkim**
*Issue #, if available:*
[319](https://github.com/aws-cloudformation/cloudformation-cli-java-plugin/issues/319)

*Description of changes:*

This change adds logging of stack id, logical resource id and response payload to aid in troubleshooting.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/main/java/software/amazon/cloudformation/AbstractWrapper.java [hunk 1]
```diff
@@ -383,6 +385,12 @@ protected void writeResponse(final OutputStream outputStream, final ProgressEven
         }
 
         String output = this.serializer.serialize(response);
+        if (response.getStatus() != OperationStatus.IN_PROGRESS) {
+            // if status is not IN_PROGRESS, it means the response has been sanitized
+            final String logicalResourceId = getLogicalResourceId(request);
+            final String stackId = getStackId(request);
+            this.log(String.format("StackId=%s LogicalResourceId=%s Response=%s", stackId, logicalResourceId, output));
```
Comment (ammokhov) [inline src/main/java/software/amazon/cloudformation/AbstractWrapper.java:392]: should we log whole request?

Comment (wbingli) [reply src/main/java/software/amazon/cloudformation/AbstractWrapper.java:392]: Request is not sanitized. We potentially can add the request as well after sanitized the request (e.g. sanitized the model and also other metadata). 

But I think we need to be a little bit more careful to add it step by step.  Besides, we should add the request log at the beginning of the handler. 

I would prefer to skip the log request for this PR and we can add it further.

File: src/main/java/software/amazon/cloudformation/AbstractWrapper.java [hunk 2]
```diff
@@ -372,7 +372,9 @@ public void processRequest(final InputStream inputStream, final OutputStream out
 
     }
 
-    protected void writeResponse(final OutputStream outputStream, final ProgressEvent<ResourceT, CallbackT> response)
+    protected void writeResponse(final OutputStream outputStream,
```
Comment (wbingli) [inline src/main/java/software/amazon/cloudformation/AbstractWrapper.java:375]: This seems a break change for the generated code.   I would suggest you do following to keep backward compatible.  And also update the code generate to use new method. In such way,  it won't break old code if still use old plugin, but it just won't have additional logging data. 

```
protected void writeResponse(final OutputStream outputStream,final ProgressEvent<ResourceT, CallbackT> response) {
 this(outputStream, response, null);
}

**Review by wbingli — APPROVED** [review]
Thanks

**Review by ammokhov — APPROVED** [review]

**Pull request merged**